react/features/base/profile/reducer.js

```diff
@@ -51,3 +39,50 @@ ReducerRegistry.register(STORE_NAME, (state = DEFAULT_STATE, action) => {
 
     return state;
 });
+
+/**
+ * Inits the profile object based on what information we have available.
+ * Info taken into consideration:
+ *   - Old Settings.js style data
+ *   - Redux data -> pre-flattening
+ *   - Redux data -> post-flattening.
+ *
+ * @private
+ * @param {Object} state - The current state of the feature.
+ * @returns {Object}
+ */
+function _initProfile(state) {
+    // Post flattening
+    let storedProfile = state || DEFAULT_STATE;
+
+    if (state && typeof state === 'object') {
+        // Pre-flattening:
+        // In an enterprise/internal build of Jitsi Meet for Android and iOS
+        // we had base/profile's state as an object with property profile.
+        const { profile } = state;
+
+        if (profile && typeof profile === 'object') {
+            storedProfile = { ...profile };
+        }
+    }
+
+    // Old Settings.js style settings
+    const email = unescape(window.localStorage.getItem('email') || '');
+    let avatarID = unescape(window.localStorage.getItem('avatarId') || '');
```

Comment:
I'm a little confused on this part. Why are you getting values from localStorage here? I thought the persistence layer would be filling this in. Are you trying to preserve backwards compatibility of some sort?